daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala

```diff
@@ -298,231 +299,217 @@ class Runner(
         Set.empty,
       )
 
-    // Removing the early return only makes this harder to read.
-    @SuppressWarnings(Array("org.wartremover.warts.Return"))
-    def stepToValue(): Either[RuntimeException, Unit] = {
-      while (!machine.isFinal) {
+    @tailrec
+    def stepToValue(): Future[SValue] =
```

Comment:
Why using `Either` when you can use direction `Future` ?

Reply:
Because this is a synchronous call. Returning a `Future` is just confusing.

Reply:
faire enough.